src/and_then_filter.rs

```diff
@@ -6,6 +6,50 @@
 
 /// Extension trait for `Iterator<Item = Result<O, E>>` to selectively transform and map Oks and Errors.
 pub trait AndThenFilter<O, E>: Sized {
+    /// Equivalent to [Iterator::filter_map] on all `Ok` values.
+    /// The filter function can fail with a result and turn an
+    /// [Result::Ok] into a [Result::Err]
+    /// ```
+    /// use std::str::FromStr;
+    /// use resiter::and_then_filter::AndThenFilter;
+    ///
+    /// let filter_mapped: Vec<_> = vec![
+    ///     Ok("1"),
+    ///     Err("2".to_owned()),
+    ///     Ok("a"), // will become an error
+    ///     Err("4".to_owned()),
+    ///     Ok("5"), // will be filtered out
+    ///     Err("b".to_owned()),
+    ///     Err("8".to_owned()),
+    /// ]
+    /// .into_iter()
+    /// .and_then_filter(|txt| {
+    ///     let r = usize::from_str(txt).map_err(|e| e.to_string());
+    ///     match r {
```

Comment:
I think it would be okay to collapse these two lines into one:

```suggestion
    ///     match usize::from_str(txt).map_err(|e| e.to_string()) {
```